openstack/identity/identity.go

```diff
@@ -209,6 +212,7 @@ func (h Handler) checkToken(r *http.Request, tenant string, tokenGetResult getRe
 	/* TODO Caching or PKI */
 	for _, a := range h.ValidAdmins {
 		if validateProjectRole(tokenGetResult, a.Project, a.Role) == true {
+			context.Set(r, service.PrivKey, "true")
 			return true
```

Comment:
we'd do it in a different place, but similarly. We'd need to use a different Key value. This PR shows you how I'd do it - I'll submit a follow on PR to actually store the tenant ID.